In [1]:
import pandas as pd

In [2]:
chicago = pd.read_csv("chicago.csv")
chicago["Department"] = chicago["Department"].astype("category")
chicago.head(3)

Name    Position Title   Department Employee Annual Salary
0    AARON,  ELVIA J  WATER RATE TAKER  WATER MGMNT              $90744.00
1  AARON,  JEFFERY M    POLICE OFFICER       POLICE              $84450.00
2     AARON,  KARINA    POLICE OFFICER       POLICE              $84450.00

In [6]:
chicago.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32063 entries, 0 to 32062
Data columns (total 4 columns):
Name                      32062 non-null object
Position Title            32062 non-null object
Department                32062 non-null category
Employee Annual Salary    32062 non-null object
dtypes: category(1), object(3)
memory usage: 784.4+ KB


In [7]:
chicago["Department"].nunique()

35

In [8]:
chicago["Department"].count()

32062

# RATHER THAN WORKING ON 32062 ROWS YOU CAN SIMPLY STORE 35 UNIQUE CATEGORIES,SAVES MEMORY USAGE

In [16]:
chicago["Department"] = chicago["Department"].astype("category")

In [17]:
chicago.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32063 entries, 0 to 32062
Data columns (total 4 columns):
Name                      32062 non-null object
Position Title            32062 non-null object
Department                32062 non-null category
Employee Annual Salary    32062 non-null object
dtypes: category(1), object(3)
memory usage: 784.4+ KB


 # LOWER UPPER TITLE LEN METHOD

In [18]:
"hello world".lower()

'hello world'

In [19]:
"hello world".upper()

'HELLO WORLD'

In [20]:
"hello world".title()

'Hello World'

In [22]:
len("hello world")

11

In [23]:
chicago.head()

Name            Position Title        Department  \
0      AARON,  ELVIA J          WATER RATE TAKER       WATER MGMNT   
1    AARON,  JEFFERY M            POLICE OFFICER            POLICE   
2       AARON,  KARINA            POLICE OFFICER            POLICE   
3  AARON,  KIMBERLEI R  CHIEF CONTRACT EXPEDITER  GENERAL SERVICES   
4  ABAD JR,  VICENTE M         CIVIL ENGINEER IV       WATER MGMNT   

  Employee Annual Salary  
0              $90744.00  
1              $84450.00  
2              $84450.00  
3              $89880.00  
4             $106836.00

In [26]:
chicago["Name"].str.lower()

0                 aaron,  elvia j
1               aaron,  jeffery m
2                  aaron,  karina
3             aaron,  kimberlei r
4             abad jr,  vicente m
5                 abarca,  anabel
6               abarca,  emmanuel
7               abascal,  reece e
8            abbasi,  christopher
9           abbatacola,  robert j
10          abbatemarco,  james j
11               abbate,  terry m
12               abbott,  betty l
13              abbott,  lynise m
14         abbruzzese,  william j
15                abdallah,  zaid
16          abdelhadi,  abdalmahd
17            abdellatif,  aref r
18             abdelmajeid,  aziz
19            abdollahzadeh,  ali
20       abdul-karim,  muhammad a
21            abdullah,  daniel n
22               abdullah,  kevin
23           abdullah,  lakenya n
24            abdullah,  rashad j
25           abdulsattar,  mudhar
26           abdul-shakur,  tahir
27         abdulwahab,  abuubaida
28              abejero,  jason v
29        aber

In [27]:
chicago["Name"].str.upper()

0                 AARON,  ELVIA J
1               AARON,  JEFFERY M
2                  AARON,  KARINA
3             AARON,  KIMBERLEI R
4             ABAD JR,  VICENTE M
5                 ABARCA,  ANABEL
6               ABARCA,  EMMANUEL
7               ABASCAL,  REECE E
8            ABBASI,  CHRISTOPHER
9           ABBATACOLA,  ROBERT J
10          ABBATEMARCO,  JAMES J
11               ABBATE,  TERRY M
12               ABBOTT,  BETTY L
13              ABBOTT,  LYNISE M
14         ABBRUZZESE,  WILLIAM J
15                ABDALLAH,  ZAID
16          ABDELHADI,  ABDALMAHD
17            ABDELLATIF,  AREF R
18             ABDELMAJEID,  AZIZ
19            ABDOLLAHZADEH,  ALI
20       ABDUL-KARIM,  MUHAMMAD A
21            ABDULLAH,  DANIEL N
22               ABDULLAH,  KEVIN
23           ABDULLAH,  LAKENYA N
24            ABDULLAH,  RASHAD J
25           ABDULSATTAR,  MUDHAR
26           ABDUL-SHAKUR,  TAHIR
27         ABDULWAHAB,  ABUUBAIDA
28              ABEJERO,  JASON V
29        ABER

In [28]:
chicago["Name"].str.title()

0                 Aaron,  Elvia J
1               Aaron,  Jeffery M
2                  Aaron,  Karina
3             Aaron,  Kimberlei R
4             Abad Jr,  Vicente M
5                 Abarca,  Anabel
6               Abarca,  Emmanuel
7               Abascal,  Reece E
8            Abbasi,  Christopher
9           Abbatacola,  Robert J
10          Abbatemarco,  James J
11               Abbate,  Terry M
12               Abbott,  Betty L
13              Abbott,  Lynise M
14         Abbruzzese,  William J
15                Abdallah,  Zaid
16          Abdelhadi,  Abdalmahd
17            Abdellatif,  Aref R
18             Abdelmajeid,  Aziz
19            Abdollahzadeh,  Ali
20       Abdul-Karim,  Muhammad A
21            Abdullah,  Daniel N
22               Abdullah,  Kevin
23           Abdullah,  Lakenya N
24            Abdullah,  Rashad J
25           Abdulsattar,  Mudhar
26           Abdul-Shakur,  Tahir
27         Abdulwahab,  Abuubaida
28              Abejero,  Jason V
29        Aber

In [32]:
chicago["Position Title"] = chicago["Position Title"].str.title()

In [34]:
chicago.head(3)

Name    Position Title   Department Employee Annual Salary
0    AARON,  ELVIA J  Water Rate Taker  WATER MGMNT              $90744.00
1  AARON,  JEFFERY M    Police Officer       POLICE              $84450.00
2     AARON,  KARINA    Police Officer       POLICE              $84450.00

In [35]:
chicago["Department"]= chicago["Department"].str.title()

In [37]:
chicago.head(3)

Name    Position Title   Department Employee Annual Salary
0    AARON,  ELVIA J  Water Rate Taker  Water Mgmnt              $90744.00
1  AARON,  JEFFERY M    Police Officer       Police              $84450.00
2     AARON,  KARINA    Police Officer       Police              $84450.00

In [38]:
chicago["Name"] = chicago["Name"].str.title()

In [39]:
chicago.head(3)

Name    Position Title   Department Employee Annual Salary
0    Aaron,  Elvia J  Water Rate Taker  Water Mgmnt              $90744.00
1  Aaron,  Jeffery M    Police Officer       Police              $84450.00
2     Aaron,  Karina    Police Officer       Police              $84450.00

# to get length of each value use the len method below:

In [42]:
chicago["Department"].str.len()

0        11.0
1         6.0
2         6.0
3        16.0
4        11.0
5        12.0
6        13.0
7         4.0
8        12.0
9         8.0
10        4.0
11        6.0
12       16.0
13        6.0
14        4.0
15        6.0
16        6.0
17        4.0
18        6.0
19        4.0
20       11.0
21        4.0
22        4.0
23        4.0
24       16.0
25       11.0
26       13.0
27       16.0
28        6.0
29        4.0
         ... 
32033     6.0
32034     6.0
32035     6.0
32036    13.0
32037     4.0
32038    11.0
32039     4.0
32040     6.0
32041     4.0
32042    16.0
32043    13.0
32044     6.0
32045     4.0
32046     7.0
32047     6.0
32048     3.0
32049     4.0
32050    11.0
32051     8.0
32052     6.0
32053     4.0
32054     6.0
32055     6.0
32056    16.0
32057    16.0
32058     6.0
32059     6.0
32060     6.0
32061     4.0
32062     NaN
Name: Department, Length: 32063, dtype: float64

In [43]:
len(chicago["Department"])

32063

In [55]:
chicago = pd.read_csv("chicago.csv").dropna(how = "all")
chicago["Department"] = chicago["Department"].astype("category")
chicago.head(3)

Name    Position Title   Department Employee Annual Salary
0    AARON,  ELVIA J  WATER RATE TAKER  WATER MGMNT              $90744.00
1  AARON,  JEFFERY M    POLICE OFFICER       POLICE              $84450.00
2     AARON,  KARINA    POLICE OFFICER       POLICE              $84450.00

# .STR.REPLACE() METHOD

In [56]:
"hello world".replace("l","!")

'he!!o wor!d'

In [57]:
chicago.tail(3)

Name           Position Title Department  \
32059    ZYMANTAS,  MARK E           POLICE OFFICER     POLICE   
32060  ZYRKOWSKI,  CARLO E           POLICE OFFICER     POLICE   
32061  ZYSKOWSKI,  DARIUSZ  CHIEF DATA BASE ANALYST       DoIT   

      Employee Annual Salary  
32059              $84450.00  
32060              $87384.00  
32061             $113664.00

In [58]:
chicago["Department"]

0             WATER MGMNT
1                  POLICE
2                  POLICE
3        GENERAL SERVICES
4             WATER MGMNT
5            CITY COUNCIL
6           STREETS & SAN
7                    OEMC
8            CITY COUNCIL
9                AVIATION
10                   FIRE
11                 POLICE
12       FAMILY & SUPPORT
13                 POLICE
14                   IPRA
15                 POLICE
16                 POLICE
17                   FIRE
18                 POLICE
19                   FIRE
20            WATER MGMNT
21                   FIRE
22                   FIRE
23                   OEMC
24       GENERAL SERVICES
25            WATER MGMNT
26          STREETS & SAN
27       FAMILY & SUPPORT
28                 POLICE
29                   FIRE
               ...       
32032    GENERAL SERVICES
32033              POLICE
32034              POLICE
32035              POLICE
32036       STREETS & SAN
32037                FIRE
32038         WATER MGMNT
32039       

In [62]:
chicago["Department"]= chicago["Department"].str.replace("MGMNT","MANAGEMENT")

In [66]:
chicago["Name"] = chicago["Name"].str.replace(",", " ")

In [67]:
chicago.head()

Name            Position Title        Department  \
0      AARON   ELVIA J          WATER RATE TAKER  WATER MANAGEMENT   
1    AARON   JEFFERY M            POLICE OFFICER            POLICE   
2       AARON   KARINA            POLICE OFFICER            POLICE   
3  AARON   KIMBERLEI R  CHIEF CONTRACT EXPEDITER  GENERAL SERVICES   
4  ABAD JR   VICENTE M         CIVIL ENGINEER IV  WATER MANAGEMENT   

  Employee Annual Salary  
0              $90744.00  
1              $84450.00  
2              $84450.00  
3              $89880.00  
4             $106836.00

In [70]:
chicago["Employee Annual Salary"]=chicago["Employee Annual Salary"].str.replace("$","").astype(float)

In [71]:
chicago.head(3)

Name    Position Title        Department  \
0    AARON   ELVIA J  WATER RATE TAKER  WATER MANAGEMENT   
1  AARON   JEFFERY M    POLICE OFFICER            POLICE   
2     AARON   KARINA    POLICE OFFICER            POLICE   

   Employee Annual Salary  
0                 90744.0  
1                 84450.0  
2                 84450.0

# YOU HAVE TO REMOVE $ SIGN FROM THE ANNUAL SALARY COLUMN AS MATHEMATICAL FUNCTIONS WILL GO OUT OF THE WINDOW SO WE NEED TO CONVERT TO FLOATING NUMBERS

# mean median etc maths function dont work on string

In [72]:
chicago["Employee Annual Salary"].sum()

2571506375.36

In [74]:
chicago["Employee Annual Salary"].mean()

80204.178633899

In [75]:
chicago["Employee Annual Salary"].std()

25098.329867510587

In [78]:
chicago["Employee Annual Salary"].nlargest(3)

8184     300000.0
7954     216210.0
25532    202728.0
Name: Employee Annual Salary, dtype: float64

In [81]:
chicago["Employee Annual Salary"].nsmallest(3)

15102       0.96
12       2756.00
27       2756.00
Name: Employee Annual Salary, dtype: float64

# FILTERING WITH STRING METHODS()

In [83]:
chicago = pd.read_csv("chicago.csv").dropna(how = "all")
chicago["Department"] = chicago["Department"].astype("category")
chicago.head(3)

Name    Position Title   Department Employee Annual Salary
0    AARON,  ELVIA J  WATER RATE TAKER  WATER MGMNT              $90744.00
1  AARON,  JEFFERY M    POLICE OFFICER       POLICE              $84450.00
2     AARON,  KARINA    POLICE OFFICER       POLICE              $84450.00

# TO FIND IN POSITION TITLE ALL THE VALUES WHERE THE WORD WATER EXISTS - BUT FIRST ITS A GOOD PRACTICE TO MAKE EVERYTHING LOWER CASE SO PANDAS CAN SUPPORT THE SAME FORMAT 

In [85]:
chicago["Position Title"].str.lower()

0                                     water rate taker
1                                       police officer
2                                       police officer
3                             chief contract expediter
4                                    civil engineer iv
5                                 asst to the alderman
6                                general laborer - dss
7                          traffic control aide-hourly
8                           staff asst to the alderman
9                                  electrical mechanic
10                                   fire engineer-emt
11                                      police officer
12                                  foster grandparent
13                                           clerk iii
14                              investigator - ipra ii
15                                      police officer
16                                      police officer
17       firefighter (per arbitrators award)-paramedic
18        

# if there is water lower and Water with w capital and WATER all capital - pandas will not be able to work on all the values where the word water exists

# NOW ITS SAFE TO SEARCH FOR WATER IN ALL LOWERCASE(m)

# IT WILL SEARCH WATER IN THE COLUMN NO MATTER IF ITS AT THE BEGINNING MIDDLE OR END

In [91]:
find_water = chicago["Position Title"].str.lower().str.contains("water")

In [94]:
chicago[find_water].head()

Name                                 Position Title  \
0        AARON,  ELVIA J                               WATER RATE TAKER   
554   ALUISE,  VINCENT G             FOREMAN OF WATER PIPE CONSTRUCTION   
671      ANDER,  PERRY A                               WATER CHEMIST II   
685  ANDERSON,  ANDREW J  DISTRICT SUPERINTENDENT OF WATER DISTRIBUTION   
702    ANDERSON,  DONALD             FOREMAN OF WATER PIPE CONSTRUCTION   

      Department Employee Annual Salary  
0    WATER MGMNT              $90744.00  
554  WATER MGMNT             $102440.00  
671  WATER MGMNT              $82044.00  
685  WATER MGMNT             $109272.00  
702  WATER MGMNT             $102440.00

# SEARCH FOR WATER AT THE BEGINNING 

In [96]:
position_start_with_water = chicago["Position Title"].str.lower().str.startswith("water")

In [99]:
chicago[position_start_with_water].head()

Name    Position Title   Department Employee Annual Salary
0        AARON,  ELVIA J  WATER RATE TAKER  WATER MGMNT              $90744.00
671      ANDER,  PERRY A  WATER CHEMIST II  WATER MGMNT              $82044.00
1054    ASHLEY,  KARMA T  WATER CHEMIST II  WATER MGMNT              $82044.00
1079   ATKINS,  JOANNA M  WATER CHEMIST II  WATER MGMNT              $82044.00
1181  AZEEM,  MOHAMMED A  WATER CHEMIST II  WATER MGMNT              $53172.00

In [101]:
ends_with_water =chicago["Position Title"].str.lower().str.endswith("ist")

In [104]:
chicago[ends_with_water].head()

Name                        Position Title  \
184       AFROZ,  NAYYAR                          PSYCHIATRIST   
308     ALARCON,  LUIS J            LOAN PROCESSING SPECIALIST   
422     ALLAIN,  CAROLYN  SENIOR TELECOMMUNICATIONS SPECIALIST   
472       ALLEN,  ROBERT                             MACHINIST   
705  ANDERSON,  EDWARD M             SR PROCUREMENT SPECIALIST   

                Department Employee Annual Salary  
184                 HEALTH              $99840.00  
308  COMMUNITY DEVELOPMENT              $81948.00  
422                   DoIT              $89880.00  
472            WATER MGMNT              $94328.00  
705            PROCUREMENT              $91476.00

# LSTRIP, STRIP AND RSTRIP

In [9]:
chicago = pd.read_csv("chicago.csv").dropna(how = "all")
chicago["Department"] = chicago["Department"].astype("category")
chicago.head(3)

Name    Position Title   Department Employee Annual Salary
0    AARON,  ELVIA J  WATER RATE TAKER  WATER MGMNT              $90744.00
1  AARON,  JEFFERY M    POLICE OFFICER       POLICE              $84450.00
2     AARON,  KARINA    POLICE OFFICER       POLICE              $84450.00

In [10]:
"    hello world".lstrip()

'hello world'

# STRIPS SPACES FROM LEFT 

In [11]:
"hello world    ".rstrip()

'hello world'

In [12]:
"   hello world   ".strip()

'hello world'

In [15]:
chicago["Name"] = chicago["Name"].str.lstrip().str.rstrip()

In [16]:
chicago.head(3)

Name    Position Title   Department Employee Annual Salary
0    AARON,  ELVIA J  WATER RATE TAKER  WATER MGMNT              $90744.00
1  AARON,  JEFFERY M    POLICE OFFICER       POLICE              $84450.00
2     AARON,  KARINA    POLICE OFFICER       POLICE              $84450.00

In [17]:
chicago["Position Title"] = chicago["Position Title"].str.strip()

In [18]:
chicago.head()

Name            Position Title        Department  \
0      AARON,  ELVIA J          WATER RATE TAKER       WATER MGMNT   
1    AARON,  JEFFERY M            POLICE OFFICER            POLICE   
2       AARON,  KARINA            POLICE OFFICER            POLICE   
3  AARON,  KIMBERLEI R  CHIEF CONTRACT EXPEDITER  GENERAL SERVICES   
4  ABAD JR,  VICENTE M         CIVIL ENGINEER IV       WATER MGMNT   

  Employee Annual Salary  
0              $90744.00  
1              $84450.00  
2              $84450.00  
3              $89880.00  
4             $106836.00

In [20]:
chicago = pd.read_csv("chicago.csv",index_col="Name").dropna(how = "all")
chicago["Department"] = chicago["Department"].astype("category")
chicago.head(3)

Position Title   Department Employee Annual Salary
Name                                                                   
AARON,  ELVIA J    WATER RATE TAKER  WATER MGMNT              $90744.00
AARON,  JEFFERY M    POLICE OFFICER       POLICE              $84450.00
AARON,  KARINA       POLICE OFFICER       POLICE              $84450.00

# STRING METHODS ON INDEX & COLUMNS

In [22]:
chicago.index = chicago.index.str.strip().str.title()

In [24]:
chicago.head(3)

Position Title   Department Employee Annual Salary
Name                                                                   
Aaron,  Elvia J    WATER RATE TAKER  WATER MGMNT              $90744.00
Aaron,  Jeffery M    POLICE OFFICER       POLICE              $84450.00
Aaron,  Karina       POLICE OFFICER       POLICE              $84450.00

In [25]:
chicago.columns.str.upper()

Index(['POSITION TITLE', 'DEPARTMENT', 'EMPLOYEE ANNUAL SALARY'], dtype='object')

In [26]:
chicago.columns= chicago.columns.str.upper()

In [27]:
chicago.head(3)

POSITION TITLE   DEPARTMENT EMPLOYEE ANNUAL SALARY
Name                                                                   
Aaron,  Elvia J    WATER RATE TAKER  WATER MGMNT              $90744.00
Aaron,  Jeffery M    POLICE OFFICER       POLICE              $84450.00
Aaron,  Karina       POLICE OFFICER       POLICE              $84450.00

# SPLIT STRINGS BY CHARACTERS WITH .STR.SPLIT() METHOD 

In [42]:
chicago = pd.read_csv("chicago.csv").dropna(how = "all")
chicago["Department"] = chicago["Department"].astype("category")
chicago.head(3)

Name    Position Title   Department Employee Annual Salary
0    AARON,  ELVIA J  WATER RATE TAKER  WATER MGMNT              $90744.00
1  AARON,  JEFFERY M    POLICE OFFICER       POLICE              $84450.00
2     AARON,  KARINA    POLICE OFFICER       POLICE              $84450.00

In [43]:
"hello my name is boris".split(" ")

['hello', 'my', 'name', 'is', 'boris']

In [50]:
chicago["Name"].str.split(",").str.get(0).str.title().value_counts()

Williams         293
Johnson          244
Smith            241
Brown            185
Jones            183
Rodriguez        171
Jackson          136
Garcia           130
Davis            127
Hernandez        110
Martinez         108
Lopez            106
Gonzalez         104
Perez            100
Wilson            94
Rivera            90
Thomas            89
Anderson          82
Torres            81
Murphy            80
Robinson          79
Moore             78
Sanchez           76
Harris            76
Miller            75
Lewis             74
Taylor            73
Martin            72
White             66
Clark             66
                ... 
Matheos            1
Drey               1
Orange             1
Berry-Mahmood      1
Shukstor           1
Gibula             1
Csongradi          1
Girven             1
Ekwueme            1
Knapp              1
Kniaz              1
Schafer            1
Bajorek            1
Tompkins           1
Stritzel           1
Rykala             1
Areizaga     

# VALUE_COUNTS -  DOES A COUNT OF ALL UNIQUE VALUES AND NLARGEST IS MORE FOR NUMERIC NUMBERS/DATA.
# STR.GET(0) - IS THE INDEX POSITION OF THE COLUMN
# TITLE CAPITALIZES THE FIRST LETTER OF EACH WORD


# find the most common first name in the position title department 

# index position of each list can be called like ----> str.get(0)

In [62]:
chicago["Position Title"].str.split(" ").str.get(0).str.title().value_counts()

Police                   10856
Firefighter-Emt           1509
Sergeant                  1186
Pool                       918
Firefighter                810
Crossing                   775
Motor                      721
Sanitation                 715
Paramedic                  641
Asst                       606
Fire                       512
Traffic                    512
Senior                     470
Construction               452
Lieutenant-Emt             394
Administrative             375
Library                    365
Librarian                  335
Lieutenant                 332
Operating                  324
Electrical                 313
Aviation                   309
Firefighter/Paramedic      259
General                    257
Staff                      250
Clerk                      242
Foreman                    237
Hoisting                   214
Deputy                     213
Machinist                  210
                         ...  
Telephone                    1
Warranty

In [60]:
chicago.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 32062 entries, 0 to 32061
Data columns (total 4 columns):
Name                      32062 non-null object
Position Title            32062 non-null object
Department                32062 non-null category
Employee Annual Salary    32062 non-null object
dtypes: category(1), object(3)
memory usage: 1.0+ MB


# MORE PRACTICE WITH SPLITS

In [63]:
chicago = pd.read_csv("chicago.csv").dropna(how = "all")
chicago["Department"] = chicago["Department"].astype("category")
chicago.head(3)

Name    Position Title   Department Employee Annual Salary
0    AARON,  ELVIA J  WATER RATE TAKER  WATER MGMNT              $90744.00
1  AARON,  JEFFERY M    POLICE OFFICER       POLICE              $84450.00
2     AARON,  KARINA    POLICE OFFICER       POLICE              $84450.00

In [69]:
chicago["Name"].str.split(",").str.get(0).value_counts().head(3)

WILLIAMS    293
JOHNSON     244
SMITH       241
Name: Name, dtype: int64

# TO GET MOST COMMON FIRST NAME

In [71]:
chicago["Name"].str.split(",").str.get(1)

0              ELVIA J
1            JEFFERY M
2               KARINA
3          KIMBERLEI R
4            VICENTE M
5               ANABEL
6             EMMANUEL
7              REECE E
8          CHRISTOPHER
9             ROBERT J
10             JAMES J
11             TERRY M
12             BETTY L
13            LYNISE M
14           WILLIAM J
15                ZAID
16           ABDALMAHD
17              AREF R
18                AZIZ
19                 ALI
20          MUHAMMAD A
21            DANIEL N
22               KEVIN
23           LAKENYA N
24            RASHAD J
25              MUDHAR
26               TAHIR
27           ABUUBAIDA
28             JASON V
29              EARL S
             ...      
32032            JAMES
32033           JUAN M
32034            OSCAR
32035           RONALD
32036           THOMAS
32037          VICENTE
32038             ERIK
32039           BILL M
32040            KELLY
32041          CAROL A
32042          JEFFREY
32043            MAC Z
32044      

In [74]:
chicago["Name"].str.split(",").str.get(1).str.strip().str.split(" ").str.get(0).value_counts()

MICHAEL        1153
JOHN            899
JAMES           676
ROBERT          622
JOSEPH          537
DAVID           506
THOMAS          490
DANIEL          472
WILLIAM         397
ANTHONY         385
KEVIN           331
BRIAN           320
RICHARD         314
MARK            310
PATRICK         300
MATTHEW         247
TIMOTHY         243
JOSE            224
STEVEN          220
CHRISTOPHER     217
PAUL            205
EDWARD          205
KENNETH         201
ERIC            170
CHARLES         163
JEFFREY         152
GREGORY         150
GEORGE          149
MARY            147
RONALD          147
               ... 
AURY              1
TAURUS            1
TSHOMA            1
ARSHANETTE        1
KENHASA           1
HARDY             1
ZENIA             1
KIMBERLEI         1
KANUTRIAL         1
DUJUAN            1
JENELLE           1
EDSEL             1
SHIRIZYK          1
CHARICE           1
SHARONDA          1
ROYCE             1
SHAZADIA          1
RANJIT            1
SHUN-KEN          1


# STRIP REMOVES ANY SPACES
# SPLIT AS THE NAME SAYS SPLITS THE VALUES IN THE LIST
# VALUE_COUNTS GETS THE COUNT OF EACH VALUE IN THE LIST 

# THE EXPAND AND N PARAMETERS OF THE STR.SPLIT() METHOD

In [21]:
chicago = pd.read_csv("chicago.csv").dropna(how = "all")
chicago["Department"] = chicago["Department"].astype("category")
chicago.head(3)

Name    Position Title   Department Employee Annual Salary
0    AARON,  ELVIA J  WATER RATE TAKER  WATER MGMNT              $90744.00
1  AARON,  JEFFERY M    POLICE OFFICER       POLICE              $84450.00
2     AARON,  KARINA    POLICE OFFICER       POLICE              $84450.00

# EXPAND = TRUE MAKES THE SERIES INTO A DATAFRAME 

In [34]:
chicago[["First Name","Last Name"]] = chicago["Name"].str.split(",",expand=True).head(3)

In [35]:
chicago.head()

Name            Position Title        Department  \
0      AARON,  ELVIA J          WATER RATE TAKER       WATER MGMNT   
1    AARON,  JEFFERY M            POLICE OFFICER            POLICE   
2       AARON,  KARINA            POLICE OFFICER            POLICE   
3  AARON,  KIMBERLEI R  CHIEF CONTRACT EXPEDITER  GENERAL SERVICES   
4  ABAD JR,  VICENTE M         CIVIL ENGINEER IV       WATER MGMNT   

  Employee Annual Salary First Name    Last Name  
0              $90744.00      AARON      ELVIA J  
1              $84450.00      AARON    JEFFERY M  
2              $84450.00      AARON       KARINA  
3              $89880.00        NaN          NaN  
4             $106836.00        NaN          NaN

In [37]:
chicago["Position Title"].str.split(" ",expand=True).head(3)

0        1      2     3     4     5     6     7     8
0   WATER     RATE  TAKER  None  None  None  None  None  None
1  POLICE  OFFICER   None  None  None  None  None  None  None
2  POLICE  OFFICER   None  None  None  None  None  None  None

In [38]:
chicago.head(3)

Name    Position Title   Department Employee Annual Salary  \
0    AARON,  ELVIA J  WATER RATE TAKER  WATER MGMNT              $90744.00   
1  AARON,  JEFFERY M    POLICE OFFICER       POLICE              $84450.00   
2     AARON,  KARINA    POLICE OFFICER       POLICE              $84450.00   

  First Name    Last Name  
0      AARON      ELVIA J  
1      AARON    JEFFERY M  
2      AARON       KARINA

# DO A SPLIT => AFTER THE FIRST SPACE, EVERYTHING THAT COMES GOES TO THE SECOND COLUMN

In [33]:
chicago["Position Title"].str.split(" ",expand=True, n= 1).head(3)

0           1
0   WATER  RATE TAKER
1  POLICE     OFFICER
2  POLICE     OFFICER

In [39]:
chicago[["First title word","Second title word"]] = chicago["Position Title"].str.split(" ",expand=True, n= 1).head(3)

In [40]:
chicago.head(3)

Name    Position Title   Department Employee Annual Salary  \
0    AARON,  ELVIA J  WATER RATE TAKER  WATER MGMNT              $90744.00   
1  AARON,  JEFFERY M    POLICE OFFICER       POLICE              $84450.00   
2     AARON,  KARINA    POLICE OFFICER       POLICE              $84450.00   

  First Name    Last Name First title word Second title word  
0      AARON      ELVIA J            WATER        RATE TAKER  
1      AARON    JEFFERY M           POLICE           OFFICER  
2      AARON       KARINA           POLICE           OFFICER

In [42]:
chicago["Department"].str.split(" ",expand=True,n=1).head(3)

0      1
0   WATER  MGMNT
1  POLICE   None
2  POLICE   None

In [43]:
chicago[["Department name", "Department last name"]] = chicago["Department"].str.split(" ",expand=True,n=1).head(3)

In [46]:
chicago.head(3)

Name    Position Title   Department Employee Annual Salary  \
0    AARON,  ELVIA J  WATER RATE TAKER  WATER MGMNT              $90744.00   
1  AARON,  JEFFERY M    POLICE OFFICER       POLICE              $84450.00   
2     AARON,  KARINA    POLICE OFFICER       POLICE              $84450.00   

  First Name    Last Name First title word Second title word Department name  \
0      AARON      ELVIA J            WATER        RATE TAKER           WATER   
1      AARON    JEFFERY M           POLICE           OFFICER          POLICE   
2      AARON       KARINA           POLICE           OFFICER          POLICE   

  Department last name  
0                MGMNT  
1                 None  
2                 None